tests/jupyterhub_api/test_check_api.py

```diff
@@ -0,0 +1,82 @@
+
+
+import json
+import requests
+import subprocess
+import os
+import docker
+import sys
+sys.path.append("..")
+import time
+from logger import Logger, LOG_FOLDER, LOG_EXTENSION
+from SessionUtils import Test, JupyterhubTest
+from requests.packages.urllib3.exceptions import InsecureRequestWarning
+requests.packages.urllib3.disable_warnings(InsecureRequestWarning)
+
+import argparse
+
+def get_args():
+    parser = argparse.ArgumentParser(description='Arguments', formatter_class = argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument("--port", dest="port", type=int,
+                        required = True,
+                        )
+
+    parser.add_argument("--token", dest="token",
+                        required=False,
+                        help='token')
+
+    parser.add_argument("--base_path", dest="base",
+                        required=False,
+                        help='base path')
+
+    args = parser.parse_args()
+    return args
+
+
+class CheckAPI(JupyterhubTest):
+    """
+    Test if a api is reachable
+
+    """
+    def __init__(self, port, token, base_path, verify):
+        param={}
```

Comment:
I could use it as an instance variable, but then I would need to access it before calling parent's constructor, which is :
```
 def _init_(self, port, token, base_path, verify):
        self.ref_test_name = "APIReachable"
        self.param['hello'] = 'abc'
        JupyterhubTest._init_(self, port, token, base_path, verify)
```
but this would throw an attribute error, if I used it after `JupyterhubTest._init_(self, port, token, base_path, verify)` then the parent's constructor would call log_params() and no parameter will get written. A class variable would not be good as parameters of other child classes would also get appended to the dictionary. Maybe I can do `self.param = {'hello' : 'abc'}` and then pass self.param to the constructor but then it would be the same thing I was dong previously.

Reply:
> A class variable would not be good as parameters of other child classes would also get appended to the dictionary. 

What do you mean? Objects are independent. If you declare a class variable, each instance has its own so there is no sharing between different childs...

Reply:
Sorry, you were right. Pass in the constructor of the parent.